azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -121,6 +121,16 @@ func SchemaDefaultNodePool() *schema.Schema {
 					ValidateFunc: validation.IntAtLeast(1),
 				},
 
+				"os_disk_type": {
+					Type:     schema.TypeString,
+					Optional: true,
+					ForceNew: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						"Ephemeral",
+						"Managed",
+					}, true),
```

Comment:
can we make this case-sensitive:

```suggestion
					}, false),
```